benchmark/catalog/catalog_benchmark.cpp

```diff
@@ -0,0 +1,314 @@
+    delete gc_;
+    delete txn_manager_;
+    delete deferred_action_manager_;
+    delete timestamp_manager_;
+  }
+
+  catalog::Catalog *catalog_;
+  storage::RecordBufferSegmentPool buffer_pool_{100, 100};
+  storage::BlockStore block_store_{100, 100};
+  transaction::TimestampManager *timestamp_manager_;
+  transaction::DeferredActionManager *deferred_action_manager_;
+  transaction::TransactionManager *txn_manager_;
+
+  storage::GarbageCollector *gc_;
+  catalog::db_oid_t db_;
+
+  std::pair<catalog::table_oid_t, catalog::index_oid_t> AddUserTableAndIndex() {
+    auto txn = txn_manager_->BeginTransaction();
+    auto accessor = catalog_->GetAccessor(txn, db_);
+
+    // Create the column definition (no OIDs)
+    std::vector<catalog::Schema::Column> cols;
+    cols.emplace_back("id", type::TypeId::INTEGER, false,
+                      parser::ConstantValueExpression(type::TransientValueFactory::GetNull(type::TypeId::INTEGER)));
+    cols.emplace_back("user_col_1", type::TypeId::INTEGER, false,
+                      parser::ConstantValueExpression(type::TransientValueFactory::GetNull(type::TypeId::INTEGER)));
+    auto tmp_schema = catalog::Schema(cols);
+
+    const auto table_oid = accessor->CreateTable(accessor->GetDefaultNamespace(), "test_table", tmp_schema);
+    TERRIER_ASSERT(table_oid != catalog::INVALID_TABLE_OID, "table creation should not fail");
+    auto schema = accessor->GetSchema(table_oid);
+    auto table = new storage::SqlTable(&block_store_, schema);
+
+    auto result UNUSED_ATTRIBUTE = accessor->SetTablePointer(table_oid, table);
+    TERRIER_ASSERT(result, "setting table pointer should not fail");
+
+    // Create the index
+    std::vector<catalog::IndexSchema::Column> key_cols{
+        catalog::IndexSchema::Column{"id", type::TypeId::INTEGER, false,
+                                     parser::ColumnValueExpression(db_, table_oid, schema.GetColumn("id").Oid())}};
+    auto index_schema = catalog::IndexSchema(key_cols, true, true, false, true);
+    const auto idx_oid =
+        accessor->CreateIndex(accessor->GetDefaultNamespace(), table_oid, "test_table_idx", index_schema);
+    TERRIER_ASSERT(idx_oid != catalog::INVALID_INDEX_OID, "index creation should not fail");
+    auto true_schema = accessor->GetIndexSchema(idx_oid);
+
+    storage::index::IndexBuilder index_builder;
+    index_builder.SetOid(idx_oid).SetKeySchema(true_schema).SetConstraintType(storage::index::ConstraintType::UNIQUE);
+    auto index = index_builder.Build();
+
+    result = accessor->SetIndexPointer(idx_oid, index);
+    TERRIER_ASSERT(result, "setting index pointer should not fail");
+
+    txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+
+    return {table_oid, idx_oid};
+  }
+};
+
+// NOLINTNEXTLINE
+BENCHMARK_DEFINE_F(CatalogBenchmark, GetAccessor)(benchmark::State &state) {
+  auto *txn = txn_manager_->BeginTransaction();
+
+  // NOLINTNEXTLINE
+  for (auto _ : state) {
```

Comment:
I guess a general question I have is how do these benchmarks cover benchmarking future catalog caching. In particular, how many times does `for (auto _ : state) {...}` execute? Is that something set by us or gbenchmark? Is it deterministic?

Reply:
It is not deterministic, really. It's as many times as Google Benchmark needs to get a stable number. In our case, it will likely report the cached performance since it'll likely run enough iterations for it to report the stable, cached number.